### PR TITLE
Alters X-ray rifle to swap modes with unique action

### PR DESCRIPTION
**Pull request by Losenis**
## About The Pull Request
Gives X-ray rifle the ability to swap modes with unique action rather than using a radial menu.

## Why It's Good For The Game
The rifle had 2 modes, so if you wanted to swap you'd switch to the other mode rather than picking from >2 like the laser pistol. Now hitting spacebar will swap modes on the fly, which for the xray rifle is very important due to the ability to change to a mode that ignores walls.

https://cdn.discordapp.com/attachments/606917654812295190/942475801717248030/2022-02-13_14-37-03.webm
image

## Changelog
:cl:
add: X-Ray laser rifle can now swap modes on the fly with Unique Action (default: spacebar)
/:cl:

### CODE REVIEW

**Comment by Luxyboo** [comment]
pretty cool, you think you can do this with the other las guns?

**Comment by hyper2snyper** [comment]
Don't make a new proc, just make the other lasgun code switch between the options when the firemode list is 2

**Comment by Losenis** [comment]
Tell splinter that, he's the guy that helped me because I didn't know how to do that

**Comment by Araulius** [comment]
Oh *please* merge this, it would make the thing so much more viable.

File: code/modules/projectiles/guns/energy.dm [hunk 1]
```diff
@@ -791,3 +792,37 @@
...
+		mode_index = 1
+	var/fake_choice = mode_list[mode_index]
+	var/datum/lasrifle/base/choice = mode_list[fake_choice]
+	
+	gun_firemode = initial(choice.fire_mode)
+	ammo_datum_type = initial(choice.ammo_datum_type)
+	fire_delay = initial(choice.fire_delay)
+	burst_amount = initial(choice.burst_amount)
+	fire_sound = initial(choice.fire_sound)
+	SEND_SIGNAL(src, COMSIG_GUN_BURST_SHOTS_TO_FIRE_MODIFIED, burst_amount)
```
Comment (TiviPlus) [inline code/modules/projectiles/guns/energy.dm:810]: This code has major issues ping me on discord about it

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by TiviPlus** [comment]
Author was made aware of issues in disc, did not adress, closing as stale

**Pull request closed without merging**